src/3.2-quote-generation-task/QuoteGenerator.jsx

```diff
@@ -27,6 +27,12 @@ function QuoteGenerator() {
       category: quoteCategory === "all" ? null : quoteCategory,
     });
 
+    if (quote.error) {
+      setIsLoading(false);
+      alert('something went wrong')
+      return
+    }
```

Comment:
In test you are mocking rejected value, in that case you can just wrap fetching in try / catch to catch and handle the error

E.g.
```jsx
try {
    const quote = await fetchRandomQuote({
      category: quoteCategory === "all" ? null : quoteCategory,
    });
    // ... handle success
} catch(error) {
    // ... handle error
}
```

Reply:
Also, test expects alert element in the DOM, so you actually need to render it and for that you probably would need an additional state

Reply:
Finally managed to pass the test with rendering the alert like this: 
```js
<span role="alert">
    something went wrong
</span>
```